File: Practice_folder/DSAa/linearsearch.cpp
//linear search
#include<iostream>
using namespace std;
int linearsearch(int arr[],int size,int k)
{ 
    for(int i=0;i<size;i++)
    {
        if(arr[i]==k)
        {
            return i;
        }
    }
    return -1;

}
int main()
{
    int arr[5]={1,2,3,4,5};
    int size=5;
    int k;
    cout<<"enter the value you want to search"<<endl;
    cin>>k;
   /* bool ans=linearsearch(arr,size,k );
    if(ans==true)
    {
        cout<<"element found"<<endl;
    }
    else
    {
        cout<<"element not found"<<endl;
    }*/
   linearsearch(arr,size,k);
   if(linearsearch(arr,size,k)==-1)
   {
    cout<<"element not found"<<endl;
   }
   else{
    cout<<"element found at index "<<linearsearch(arr,size,k)<<endl;
   }

}

File: java/Java programs/Linearsearch.java
public class Linearsearch 
{
    public int linearSearch(int[] arr, int target) {
        
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i; 
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        LinearSearch search = new LinearSearch();

        
        int[] numbers = {10, 20, 30, 40, 50};


        int target = 30;

    
        int result = search.linearSearch(numbers, target);
        
        if (result != -1) {
            System.out.println("Element found at index: " + result);
        } else {
            System.out.println("Element not found");
        }
    }
}


